Compiling lv_log.c fails due to type limits error

Hi!

When compiling LittlevGL with the GCC 7.3.1 provided by Arm (`arm-bare-none-eabi`) with `-Wall`, `-Wextra` and `-Werror` activated and `LV_LOG_LEVEL` set to `LV_LOG_LEVEL_TRACE`, compilation fails with the following error

```
lv_log.c:61:14: error: comparison is always true due to limited range of data type [-Werror=type-limits]
     if(level >= LV_LOG_LEVEL) {
              ^~
cc1: all warnings being treated as errors
```

because `level` is a `lv_log_level_t` (which is a typedef for `uint8_t`) and `LV_LOG_LEVEL_TRACE` equals `0`.

There are multiple possibilities how to fix this (in case you think that this is an error, too):

* Start enumerating log levels with `1`
* Use a non-unsigned integer type for `lv_log_level_t`
* Enumerate log levels in a way that bitwise comparisons could be done
* Streamline the definitions of the log macros in a way that the comparison is completely superfluous (though this would imply that `lv_log_add()` is kinda internal and shall be used _exclusively_ through the log macros)
* Add compiler pragmas for "downgrading" this to a warning
* Your obvious solution that I didn't think of

I'd gladly provide a patch for any of the above if you like and – again – if you think this is an issue, too, otherwise feel free to simply close this issue.

Regards

--- a/lv_misc/lv_log.h
+++ b/lv_misc/lv_log.h
@@ -32,7 +32,7 @@ extern "C" {
 #define LV_LOG_LEVEL_ERROR 3     /*Only critical issue, when the system may fail*/
 #define _LV_LOG_LEVEL_NUM  4
 
-typedef uint8_t lv_log_level_t;
+typedef int8_t lv_log_level_t;
 
 #if USE_LV_LOG
 /**********************

